superduperdb/container/model.py

```diff
@@ -353,34 +354,18 @@ def _predict_with_select_and_ids(
 
         outputs = self.predict(X=X_data, one=False, distributed=False, **kwargs)
 
-        if self.flatten:
-            assert all([isinstance(x, (list, tuple)) for x in outputs])
-
         if isinstance(self.encoder, Encoder):
-            if self.flatten:
-                outputs = [
-                    [self.encoder(x).encode() for x in output] for output in outputs
-                ]
-            else:
-                outputs = [self.encoder(x).encode() for x in outputs]
-        elif isinstance(self.output_schema, Schema):
-            encoded_ouputs = []
-            for output in outputs:
-                if isinstance(output, dict):
-                    encoded_ouputs.append(self.output_schema.encode(output))
-                elif self.flatten:
-                    encoded_output = [self.output_schema.encode(x) for x in output]
-                    encoded_ouputs.append(encoded_output)
-            outputs = encoded_ouputs if encoded_ouputs else outputs
+            outputs = [self.encoder(x).encode() for x in outputs]
+
+        # TODO encode with output-schema if supplied
 
         select.model_update(
             db=db,
             model=self.identifier,
             outputs=outputs,
             key=X,
             ids=ids,
-            document_embedded=self.model_update_kwargs.get('document_embedded', True),
-            flatten=self.flatten,
```

Comment:
flatten?

Reply:
I don't think we are actually flattening the outputs here

Reply:
Flattening happes in `model_update_kwargs`.